### DISCUSSION TITLE
Inventory Item - Create dedicated model / relationships

### DISCUSSION THREAD

**Discussion by jvanderaa** [nautobot/nautobot#1365] [Ideas]
As we worked on the original development of the device life cycle plugin, we took a look at hoping to incorporate Inventory Items into the process. In that project, we did accomplish the goal, but it was observed that Inventory Items appear to be more so just related to the particular device. There was no equivalent of `device type` in the `inventory items` section. Such as being able to report on line cards that are end of life. 

With this in mind, an idea for another model to add into Nautobot would be InventoryItemType and maybe make InventoryItem similar to that of Device with some relationships and models. 

Could also help with #126  and #1319

**Comment by Niogui** [comment]
+1
The inventory model as it exists right now is too limited. ```inventoryItemType``` would not only allow what @jvanderaa suggests in this thread, but would also provide a way to track our modules. NetBox has a Modules model (https://docs.netbox.dev/en/stable/core-functionality/modules/) in addition to the InventoryItems. We don't have the required model in Nautobot to efficiently track and manage our modules.
I am not sure that it is necessary to add a specific model for modules (in Netbox, modules are linked to devices module bays). Inventory Items might be enough to deal with that, but we definitely need a model closer to what devices are in Nautobot.

**Comment by jvanderaa** [comment]
Also relates to #2101
